SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp

```diff
@@ -326,23 +351,33 @@ void USpatialMetrics::TrackSentRPC(UFunction* Function, ERPCType RPCType, int Pa
 
 void USpatialMetrics::HandleWorkerMetrics(Worker_Op* Op)
 {
-	if (WorkerMetricsRecieved.IsBound())
+	int32 NumGuageMetrics = Op->op.metrics.metrics.gauge_metric_count;
+	int32 NumHistogramMetrics = Op->op.metrics.metrics.histogram_metric_count;
+	if (NumGuageMetrics > 0 || NumHistogramMetrics > 0)
 	{
-		int32 NumMetrics = Op->op.metrics.metrics.gauge_metric_count;
+		FString StringTmp;
+		StringTmp.Reserve(128);
 
-		if (NumMetrics > 0)
+		for (int32 i = 0; i < NumGuageMetrics; i++)
 		{
-			// Construct a map to store all the metrics and pass it to the users delegate
-			TMap<FString, double> WorkerMetrics;
-			WorkerMetrics.Reserve(NumMetrics);
+			StringTmp = Op->op.metrics.metrics.gauge_metrics[i].key;
+			WorkerSDKGaugeMetrics.FindOrAdd(StringTmp) = Op->op.metrics.metrics.gauge_metrics[i].value;
+		}
 
-			for (int32 i = 0; i < NumMetrics; i++)
+		for (int32 i = 0; i < NumHistogramMetrics; i++)
+		{
+			StringTmp = Op->op.metrics.metrics.histogram_metrics[i].key;
```

Comment:
Can you grab a reference to `Op->op.metrics.metrics.histogram_metrics[i]` up front, it'll make this code easier to parse.